examples/run_largecyc_generic_echo.py

```diff
@@ -0,0 +1,137 @@
+"""
+Spin Echo
+=========
+
+A Hahn echo is generated by feeding the generic ppg the specific ppg_list
+required. Note that the overall phase and the 90-180 phase difference are
+phase cycled in a nested way similar to run_CPMG.py. If you wish to keep
+the field as is without adjustment follow the 'py run_generic_echo.py'
+command with 'stayput' (e.g. 'py run_generic_echo.py stayput')
+"""
+
+from pylab import *
+from pyspecdata import *
+import os, sys
+from numpy import *
+import SpinCore_pp
+from SpinCore_pp import prog_plen, get_integer_sampling_intervals, save_data
+from SpinCore_pp.ppg import generic
+from datetime import datetime
+from Instruments.XEPR_eth import xepr
+
+my_exp_type = "ODNP_NMR_comp/Echoes"
+assert os.path.exists(getDATADIR(exp_type=my_exp_type))
+# {{{importing acquisition parameters
+config_dict = SpinCore_pp.configuration("active.ini")
+(
+    nPoints,
+    config_dict["SW_kHz"],
+    config_dict["acq_time_ms"],
+) = get_integer_sampling_intervals(
+    SW_kHz=config_dict["SW_kHz"],
+    time_per_segment_ms=config_dict["acq_time_ms"],
+)
+# }}}
+# {{{add file saving parameters to config dict
+config_dict["type"] = "echo"
+config_dict["date"] = datetime.now().strftime("%y%m%d")
+config_dict["echo_counter"] += 1
+# }}}
+# {{{ command-line option to leave the field untouched (if you set it once, why set it again)
+adjust_field = True
+if len(sys.argv) == 2 and sys.argv[1] == "stayput":
+    adjust_field = False
+# }}}
+input(
+    "I'm assuming that you've tuned your probe to %f since that's what's in your .ini file. Hit enter if this is true"
+    % config_dict["carrierFreq_MHz"]
+)
+# {{{ let computer set field
+if adjust_field:
+    field_G = config_dict["carrierFreq_MHz"] / config_dict["gamma_eff_MHz_G"]
+    print(
+        "Based on that, and the gamma_eff_MHz_G you have in your .ini file, I'm setting the field to %f"
+        % field_G
+    )
+    with xepr() as x:
+        assert field_G < 3700, "are you crazy??? field is too high!"
+        assert field_G > 3300, "are you crazy?? field is too low!"
+        field_G = x.set_field(field_G)
+        print("field set to ", field_G)
+# }}}
+# {{{set phase cycling
+# NOTE: The overall phase and the 90-180 phase difference are phase cycled
+# in a nested way
+ph2 = r_[0, 1, 2, 3]
```

Comment:
See comments on slack on the new way to name these.. show the phase argument above, and name them according to the conjugate coherence domain, not according to how the phase is actually cycled

Reply:
The new CPMG sequence in my pull request is an example of how to do this